data-pipeline/src/agent_info/fetcher.rs

```diff
@@ -0,0 +1,349 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+//! Provides utilities to fetch the agent /info endpoint and an automatic fetcher to keep info
+//! up-to-date
+
+use super::{schema::AgentInfo, AgentInfoArc};
+use anyhow::{anyhow, Result};
+use arc_swap::ArcSwapOption;
+use ddcommon::{connector::Connector, Endpoint};
+use hyper::{self, body::Buf, header::HeaderName};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::sleep;
+
+#[allow(clippy::declare_interior_mutable_const)]
```

Comment:
If you wanted to avoid this allow you could use `once_cell`. Something like:

```rust
use once_cell::sync::Lazy;

static DATADOG_AGENT_STATE: Lazy<HeaderName> = Lazy::new(|| HeaderName::from_static("datadog-agent-state"));

Reply:
The lint is a [false positive from clippy](https://github.com/rust-lang/rust-clippy/issues/9776) which [has been fixed in rust 1.80](https://github.com/rust-lang/rust-clippy/pull/12691) so I it is probably ok to just allow it here